homeassistant/components/bosch_alarm/__init__.py

```diff
@@ -0,0 +1,55 @@
+"""The Bosch Alarm integration."""
+
+from __future__ import annotations
+
+import asyncio
+import ssl
+
+import bosch_alarm_mode2
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_PORT, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+
+from .const import CONF_INSTALLER_CODE, CONF_USER_CODE
+from .coordinator import BoschAlarmCoordinator
+
+type BoschAlarmConfigEntry = ConfigEntry[BoschAlarmCoordinator]
+
+PLATFORMS: list[Platform] = [Platform.ALARM_CONTROL_PANEL]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: BoschAlarmConfigEntry) -> bool:
+    """Set up Bosch Alarm from a config entry."""
+    panel = bosch_alarm_mode2.Panel(
+        host=entry.data[CONF_HOST],
+        port=entry.data[CONF_PORT],
+        automation_code=entry.data.get(CONF_PASSWORD, None),
+        installer_or_user_code=entry.data.get(
+            CONF_INSTALLER_CODE, entry.data.get(CONF_USER_CODE, None)
+        ),
+    )
+
+    try:
+        await panel.connect()
+    except (PermissionError, ValueError) as err:
+        await panel.disconnect()
+        raise ConfigEntryAuthFailed from err
+    except (
+        OSError,
+        ConnectionRefusedError,
+        ssl.SSLError,
+        asyncio.exceptions.TimeoutError,
+    ) as err:
+        await panel.disconnect()
+        raise ConfigEntryNotReady("Connection failed") from err
+
+    entry.runtime_data = BoschAlarmCoordinator(hass, entry, panel)
```

Comment:
```suggestion
    coordinator = BoschAlarmCoordinator(hass, entry)
    await coordinator.async_config_entry_first_refresh()
    entry.runtime_data = coordinator
```

`async_config_entry_first_refresh` will call `_async_setup` on the coordinator and then we have the panel connection nice and tidy in there.

Reply:
`async_config_entry_first_refresh` ends up calling `_async_update_data` which bails since ive been setting `update_method` to None, as all the documentation suggests you do that for push based coordinators. Should i just not be using `async_config_entry_first_refresh` in this scenario?